src/relentless/model/potential/bond.py

```diff
@@ -0,0 +1,568 @@
+import numpy
+
+from relentless import math
+from relentless.model import variable
+
+from . import potential
+
+
+class BondParameters(potential.Parameters):
+    """Parameters for a bond potential."""
+
+    pass
+
+
+class BondPotential(potential.BondedPotential):
+    r"""Abstract base class for a bond potential."""
+
+    pass
+
+
+class HarmonicBond(BondPotential):
+    r"""Harmonic bond potential.
+
+    .. math::
+
+        u(r) = \frac{k}{2} (r - r_0)^2
+
+    where :math:`r` is the distance between two bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+
+    | Parameter   | Description                                      |
+    +=============+==================================================+
+    | ``k``       | Spring constant :math:`k`.                       |
+    +-------------+--------------------------------------------------+
+    | ``r0``      | Minimum-energy length :math:`r_0`.               |
+    +-------------+--------------------------------------------------+
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`BondParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    Harmonic Bond::
+
+        >>> u = relentless.potential.bond.Harmonic(("A"))
+        >>> u.coeff["A"].update({'k': 1000, 'r0': 1})
+
+    """
+
+    def __init__(self, types, name=None):
+        super().__init__(keys=types, params=("k", "r0"), name=name)
+
+    def energy(self, types, r):
```

Comment:
Small thing: rename to either `type_` or `bond_type` throughout since there is only one type for a bond.
```suggestion
    def energy(self, type_, r):
```
You should not use `type` itself because that is a built-in function.